nemo/collections/nlp/modules/common/megatron/token_level_encoder_decoder.py

```diff
@@ -82,142 +83,102 @@ class MegatronTokenLevelEncoderDecoderModule(MegatronModule):
         num_tokentypes=0,
         parallel_output=True,
         pre_process=True,
         post_process=True,
-        init_method_std=0.02,
         fp16_cross_entropy=False,
         use_cpu_initialization=False,
-        hidden_dropout=0.1,
-        attention_dropout=0.1,
-        position_embedding_type='learned_absolute',
-        relative_attention_num_buckets=32,
-        relative_attention_max_distance=128,
-        relative_position_bias_self_attention_only=False,
         precision=16,
-        fp32_residual_connection=False,
-        activations_checkpoint_method=None,
-        activations_checkpoint_num_layers=1,
-        layernorm_epsilon=1e-5,
-        persist_layer_norm=False,
-        bias_activation_fusion=True,
-        bias_dropout_add_fusion=True,
-        masked_softmax_fusion=True,
-        openai_gelu=False,
-        activation='gelu',
-        onnx_safe=False,
-        bias=True,
-        normalization='layernorm',
-        transformer_block_type='pre_ln',
-        hidden_steps=-1,
-        headscale=False,
+        embedding_init_method_std=0.02,
+        embedding_dropout=0.1,
         add_encoder=True,
         add_decoder=True,
-        num_self_attention_per_cross_attention=1,
         share_token_embeddings=True,
         share_decoder_tokens_head_embeddings=True,
     ):
         super(MegatronTokenLevelEncoderDecoderModule, self).__init__()
 
+        self.encoder_cfg = encoder_cfg
+        self.decoder_cfg = decoder_cfg
         self.parallel_output = parallel_output
         self.pre_process = pre_process
         self.post_process = post_process
         self.fp16_cross_entropy = fp16_cross_entropy
         self.precision = precision
         self.add_encoder = add_encoder
         self.add_decoder = add_decoder
-        self.normalization = normalization
-        self.position_embedding_type = position_embedding_type
-        self.relative_attention_num_buckets = relative_attention_num_buckets
-        self.relative_attention_max_distance = relative_attention_max_distance
         self.share_token_embeddings = share_token_embeddings
         self.share_decoder_tokens_head_embeddings = share_decoder_tokens_head_embeddings
 
-        if self.position_embedding_type == 'learned_absolute':
-            add_position_embedding = True
-        elif self.position_embedding_type == 'relative':
-            add_position_embedding = False
-        else:
-            raise ValueError('Unknown position embeeding type. Options: ' '[learned_absolute | relative]')
-        self.relative_position_bias_self_attention_only = relative_position_bias_self_attention_only
-
-        if kv_channels is None:
-            assert (
-                hidden_size % num_attention_heads == 0
-            ), 'hidden_size must be divisible by num_attention_heads if kv_channels is None'
-            kv_channels = hidden_size // num_attention_heads
+        encoder_kv_channels, decoder_kv_channels = self._validate_config()
```

Comment:
Why do we need those values `encoder_kv_channels, decoder_kv_channels`?

Reply:
Because these are typically provided as None in the yaml and then computed internally based on hidden size and num attention heads.